### PR TITLE
Update Docker image push workflow

### PR DESCRIPTION
**Pull request by ErikSchierboom**
This PR updates the Docker build-push action to v2, which has some neat improvements such as integrating with GitHub caching without needing any additional dependencies. 

We've also combined the two workflows into one workflow, which is more efficient.

### CODE REVIEW

**Comment by ErikSchierboom** [comment]
@exercism/github-actions Could you take a look at this?

**Review by cmccandless — APPROVED** [review]
LGTM. I'm not familiar with every action used here, but it looks straightforward.

**Comment by mpizenberg** [comment]
looks neat. Does it push to both dockerhub and ecr when both login are done in the same flow?

**Comment by ErikSchierboom** [comment]
Yes, it should. See this part of the docs: https://github.com/docker/build-push-action#push-to-multi-registries

**Comment by ErikSchierboom** [comment]
And caching works very nice: https://github.com/exercism/csharp-test-runner/pull/36/checks?check_run_id=1656360834

**Review by iHiD — APPROVED** [review]

File: .github/workflows/docker.yml [hunk 1]
```diff
@@ -0,0 +1,54 @@
+name: Push Docker images to DockerHub and ECR
+
+on:
+  push:
+    branches: [master]
```
Comment (iHiD) [inline .github/workflows/docker.yml:5]: Suggest `[master,main]` (or whatever the syntax is) to future-proof. I think there might even be a `${{ primary_branch }}` or something now (I totally guessed at that name)

Comment (ErikSchierboom) [reply .github/workflows/docker.yml]: It looks like I can't use any variables for the branch names, so it will be a combination.

**Comment by mpizenberg** [comment]
Did you intend to trigger on PR too before merging? I think this might result in the tests runner changing before a PR is valid

**Comment by mpizenberg** [comment]
By the way, you should have a "Re-run jobs" button in the top right corner of the detailed view of the action flow. So you don't have to push an empty commit.

**Comment by ErikSchierboom** [comment]
@mpizenberg No that was not intended. It was fixed in #38

**Pull request merged**